src/ReportingPortalAgent.php

```diff
@@ -117,7 +118,12 @@ public function beforeSuite(SuiteEvent $e)
                 }
                 $launchDescription = $this->launchDescription;
                 $this->launchDescription = preg_replace_callback($ptn, $lookupEnvVar, $launchDescription);
-                self::$httpService->launchTestRun($this->launchName, $this->launchDescription, ReportPortalHTTPService::DEFAULT_LAUNCH_MODE, $tags);
+                if ($this::$httpService::$launchID == $this::$httpService::EMPTY_ID) {
+                    self::$httpService->launchTestRun($this->launchName, $this->launchDescription, ReportPortalHTTPService::DEFAULT_LAUNCH_MODE, $tags);
```

Comment:
Why you use `$this` instead of `self`?

Reply:
It is static field (before - I did not change it)

Reply:
Actually I got something wrong (not consistent) it should be `self`